File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.models.enums import NotificationType


class NotificationResponse(BaseModel):
    id: int

    user_id: int

    title: str
    message: str

    notification_type: NotificationType

    is_read: bool

    created_at: datetime
    read_at: Optional[datetime]

    model_config = ConfigDict(from_attributes=True)


class NotificationUpdate(BaseModel):
    is_read: bool = True


class NotificationListResponse(BaseModel):
    notifications: list[NotificationResponse]

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/notification.ts
export type NotificationType =
    | "MESSAGE"
    | "MENTION"
    | "INVITE"
    | "SYSTEM";

export interface NotificationResponse {
    id: number;
    user_id: number;
    title: string;
    message: string;
    notification_type: NotificationType;
    is_read: boolean;
    created_at: string;
    read_at: string | null;
}

export interface NotificationUpdate {
    is_read: boolean;
}

export interface NotificationListResponse {
    notifications: NotificationResponse[];
}